### DISCUSSION TITLE
Share your Block Bindings use cases and showcase

### DISCUSSION THREAD

**Discussion by artemiomorales** [WordPress/gutenberg#65961] [Block Bindings]
Use this conversation to share your current or desired use cases for Block Bindings 🙌

Please feel free to discuss limitations as well — all of this will help us shape the roadmap.

**Comment by itsamoreh** [comment]
I would love to use block bindings to get meta into CPT templates.

E.g. on a "Book" CPT, the author of the book is not the author of the WP post. I would love to use a block binding to get the custom meta for "book author" to display on the template.

**Reply by SantosGuillamot** [reply]
Thanks for the feedback! 🙂 If I understand your use case correctly, this is already possible in Gutenberg, and it will be part of  6.7. In this demo, you can see how I did something really similar for a "Movie" CPT:

https://github.com/user-attachments/assets/0890a2e2-8db2-4580-ae2a-9f562b6eeff9

**Reply by roborourke** [reply]
This still doesn't really work for filtered block binding attributes and using them in the template vs in the post content. If you filter the `url` attribute of the embed block supported for bindings for example, if you use the bound embed block in the template it fails, but in the post content it works.

**Comment by jasalt** [comment]
I've been experimenting with ["Twig templating block"](https://github.com/jasalt/twig-templating-block) which allows writing custom Twig/HTML template in block editor and to map available block bindings to it's template context variables.

This allows using full templating language feature set as block bindings allow all regular PHP function return value types including collections and scalars. Block bindings seem a good fit for this use case, feeling like the "C" part in "MVC", written in code, and "V" is the part editable by user in the editor. It is also relatively safe with `unfiltered_html` required mostly equivalent to Custom HTML block with Twig auto escaping available for some extra safety.

Example binding code & template: https://gist.github.com/jasalt/dc3d9b7704da431e184096595e0637ea

Less polished Site Editor screenshot utilizing it from a site:
<img width="1741" height="815" alt="image" src="https://github.com/user-attachments/assets/61887dc0-a803-4b92-9e01-74a22161bc6f" />

Some example bindings for a video site allowing with some complex queries and processing with various 3rd party plugins, CPT's and such:

- Last Visited Videos: array of \Timber\Post objects
- Last Completed Videos: array of \Timber\Post objects
- Favourite Videos: array of \Timber\Post objects
- Season Progress Status: array {"videos_completed" => int, "videos_count" => int}
- Video Tags: array ["label" => string, "term_id" => int]
- Next Video Suggestions: array of \Timber\Post objects
- Video Visited: boolean
- Video Favourited: boolean


Looking to document it better during the autumn but the code is available for experimenting and it's quite succinct (~ 500LOC JS, ~ 400 LOC PHP).
